File: ModelGovernance/FLASKMODELGOVERNANCE/flaskr/additional_scripts/AUDIT.r
# reading commandline arguments
args = commandArgs(trailingOnly=TRUE)

# model name
model = args[1]
# hash of the dataset
hash = args[2]
# target column name
target = args[3]
# measure used in auditing
measure = args[4]
# timestamp
date = args[5]

# reading model
model = readRDS(paste0('../../../V/Models/', model, '/model'))

# reading data
data = read.csv(paste0('../../../V/Datasets/', hash), header=T)

# dropping target column
X = data[-which(colnames(data) == target)]
y = data[[target]]

# making prediction
if(class(model)[1] == 'WrappedModel') {
	# case when model is mlr model
	library(mlr)
	pred = predict(model, newdata = X)$data$response
} else if(class(model)[1] == 'train' && class(model)[2] == 'train.formula') {
	# case when model is caret model
	pred = predict(model, X)
} else if(class(model)[2] == 'model_fit') {
	# case when model is parsnip model
	pred = parsnip::predict(model, X)
}

if (measure == 'acc') {
	result = mean(pred == y)
} else if (measure == 'mae') {
	result = mean(sum(abs(pred - y)))
} else if (measure == 'mse') {
	result = mean(sum((pred - y)^2))
}

# writting result
write.table(result, paste0('../../../tmp/', date, '.txt'), row.names=FALSE, col.names=FALSE)

File: docker/ModelGovernance/FLASKMODELGOVERNANCE/flaskr/additional_scripts/AUDIT.py
"""Auditing model
"""

import warnings

warnings.filterwarnings("ignore", category=FutureWarning)

import sys
import pickle
import pandas as pd
import numpy as np

# name of the model
model = sys.argv[1]
# data hash
hash = sys.argv[2]
# target column name
target = sys.argv[3]
# measure
measure = sys.argv[4]
# timestamp
date = sys.argv[5]

# load model
with open("flaskr/V/Models/" + model + "/model", 'rb') as fd:
    model = pickle.load(fd)

# loading data

data = pd.read_csv("flaskr/V/Datasets/" + hash, delimiter=',', header=0)
y = data[target]
X = data.drop(columns=target)

# making prediction
pred = model.predict(X)

if measure == 'acc':
    result = np.mean(pred == y)
elif measure == 'mae':
    result = np.mean(np.sum(np.abs(pred - y)))
elif measure == 'mse':
    result = np.mean(np.sum((pred - y) ** 2))

# saving result
with open("flaskr/tmp/" + date + ".txt", "w") as fd:
    fd.write(str(result))
